aicsimageio/dimensions.py

```diff
@@ -121,3 +123,17 @@ def __str__(self) -> str:
 
     def __repr__(self) -> str:
         return str(self)
+
+    def __getitem__(self, key: Union[str, Sequence[str]]) -> Tuple:
+        if isinstance(key, str):
+            return (getattr(self, key),)
+        elif (
+            isinstance(key, seq)
+            and all(isinstance(k, str) for k in key)
+            and all(k in self._order for k in key)
+        ):
+            return tuple(getattr(self, k) for k in key)
+        else:
+            raise TypeError(
```

Comment:
one way to fall into this `else` is if either of the `all` checks fail, in which case the error message is not going to be as helpful.  I'm not sure whether it's worth checking and reporting those errors.  Requesting dims that don't exist.  Maybe you'd want to report the value of `key` vs `self._order`.

Reply:
Should now raise `IndexError`s for keys not tracked by the objcet

Reply:
Looks good.  This is a nice addition, thank you!